### PR TITLE
user validation

### PR DESCRIPTION
**Pull request by youngj3**
set up functional user authentication
added first and last name to migration table for users and included validations

### CODE REVIEW

**Comment by JasperGroner** [comment]
Still looks great to me!

File: server/src/db/migrations/20210101210207_createUsers.cjs [hunk 1]
```diff
@@ -17,6 +17,8 @@ exports.up = async (knex) => {
       table.bigIncrements("id");
       table.string("email").notNullable().unique();
       table.string("cryptedPassword").notNullable();
+      table.string("firstName").notNullable()
+      table.string('lastName').notNullable()
```
Comment (dovakeen118) [inline server/src/db/migrations/20210101210207_createUsers.cjs]: we should not edit existing migrations, and instead create a NEW migration file with the changes we want (like adding a new column). Remember that when the app was created and the repo was added to github that this original User migration has already been run. That means that the knex database has already looked at the timestamp of this migration file name. When we run `migrate:latest`, knex will ignore this file and thus not add the new columns you have specified.

If we leave this modification, each group member will have to drop their database on their machine, re-create the database, and run the migrations again.
Alternatively, the better solution is to always make a new migration file for any changes to tables in the database (adding tables, adding column to a table, removing a column from a table, renaming a column, etc). The changes are tacked on that new migration file, so when the PR is merged in each group member just needs to run the new migration with `migrate:latest`

**Review by dovakeen118 — COMMENTED** [review]
we need a new migration file!

File: client/src/components/App.js [hunk 2]
```diff
@@ -25,10 +25,18 @@ const App = (props) => {
...
+    message = `${currentUser.firstName}`
+  }
+  
   return (
     <Router>
       <TopBar user={currentUser} />
       <Switch>
+        <Route exact path="/">
+          <h2>Hello from react {message}</h2>
+        </Route>
```
Comment (nwalberts) [inline client/src/components/App.js]: lets just delete any sort of generic message. Just delete all of these relevant lines.

If you would like, in a future feature you can render a `LandingPage` component at the root path. For now, just get rid of it all!

File: client/src/components/registration/RegistrationForm.js [hunk 3]
```diff
@@ -89,6 +105,26 @@ const RegistrationForm = () => {
     <div className="grid-container">
       <h1>Register</h1>
       <form onSubmit={onSubmit}>
+        <div>
+          <label>First name
+            <input type="text"
+            name="firstName"
+            value={userPayload.firstName}
+            onChange={onInputChange}
```
Comment (nwalberts) [inline client/src/components/registration/RegistrationForm.js]: all props should be indented in. In this case, the input field properties (e.g. lines 111 to 113) should be indented in

File: client/src/components/registration/RegistrationForm.js [hunk 4]
```diff
@@ -89,6 +105,26 @@ const RegistrationForm = () => {
...
+        <div>
+          <label>Last name
+            <input type="text"
+            name="lastName"
+            value={userPayload.lastName}
+            onChange={onInputChange}
+            />
+            <FormError error={errors.lastName} />
+          </label>
+        </div>
```
Comment (nwalberts) [inline client/src/components/registration/RegistrationForm.js]: I know we may have merged this in previously, but the divs around each of these input dont actually serve any purpose. Delete them unless you want to target them with styling

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**